### PR TITLE
Fix kinetic build issues and travis-ci

### PR DESCRIPTION
**Pull request by jrgnicho**
This PR addresses issues that prevented the travis ci from successfully building the kinetic-devel branch, the most significant changes are:
 - removes PR2 dependencies as this repo 's maintenance has fallen behind ([see here](https://github.com/ros-planning/moveit/issues/110))
 - Builds with catkin tools
 - ran clang formatting (had to run it twice, running it once didn't catch all)

This will show lots of changes due to the clang formatting. I can break it up into two PR's, however travis ci will fail if only formatting changes are made. 
@davetcoleman please review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ This repo contains unit tests for validating various implementation of the movei
 - Install [wstool](http://wiki.ros.org/wstool) in order manage the repos inside the workspace
   ```
   sudo apt-get install python-wstool
+  sud
```
Comment (nsnitish) [inline README.md]: Is it a typo?

File: kinematics_base_test/src/test_kinematics_as_plugin.cpp [hunk 2]
```diff
@@ -1,343 +0,0 @@
- /*********************************************************************
- * Software License Agreement (BSD License)
- *
```
Comment (nsnitish) [inline kinematics_base_test/src/test_kinematics_as_plugin.cpp:3]: Why was this test_kinemtaics_as_plugin.cpp created at first place?

**Review by nsnitish — APPROVED** [review]
I think we can merge this PR with a minor change if @davetcoleman approves

File: kinematics_base_test/src/test_kinematics_as_plugin.cpp (hunk 2 above)
Comment (jrgnicho) [reply kinematics_base_test/src/test_kinematics_as_plugin.cpp:3]: This file existed in the PR2 repo at first and it was used as "inspiration" for the **test_kinematics_plugins.cpp** file.  However it was never used in this repo.

File: README.md (hunk 1 above)
Comment (jrgnicho) [reply README.md]: Yes

**Comment by davetcoleman** [comment]
Thanks for fixing all this, great contributions. Let's still merge the clang changes separate: https://github.com/ros-planning/moveit_kinematics_tests/pull/11

**Comment by davetcoleman** [comment]
My PR would not pass travis because of the other kinetic issues, I'll just go with this one.

**Pull request merged**